explorer/trace_testdata/full_trace.carbon

```diff
@@ -197,153 +197,123 @@ fn Main() -> i32 {
-// CHECK:STDOUT: (+) stack-push: (n + 1) .0.
-// CHECK:STDOUT: (+) stack-push: (n + 1) .0.
+// CHECK:STDOUT: (+) stack-push: pos-0 ValueExpressionAction `(n + 1)` (full_trace.carbon:12)
+// CHECK:STDOUT: (+) stack-push: pos-0 ExpressionAction `(n + 1)` (full_trace.carbon:12)
 // CHECK:STDOUT: --- step exp (n + 1) .0. (full_trace.carbon:12) --->
-// CHECK:STDOUT: (+) stack-push: n .0.
-// CHECK:STDOUT: (+) stack-push: n .0.
+// CHECK:STDOUT: (+) stack-push: pos-0 ValueExpressionAction `n` (full_trace.carbon:12)
+// CHECK:STDOUT: (+) stack-push: pos-0 ExpressionAction `n` (full_trace.carbon:12)
 // CHECK:STDOUT: --- step exp n .0. (full_trace.carbon:12) --->
-// CHECK:STDOUT: (-) stack-pop: n .0.
-// CHECK:STDOUT: (-) stack-pop: n .1. {{[[][[]}}0]]
+// CHECK:STDOUT: (-) stack-pop:  pos-0 ExpressionAction `n` (full_trace.carbon:12)
+// CHECK:STDOUT: (-) stack-pop:  pos-1 ValueExpressionAction `n` results: [`0`]  (full_trace.carbon:12)
 // CHECK:STDOUT: --- step exp (n + 1) .1. (full_trace.carbon:12) --->
-// CHECK:STDOUT: (+) stack-push: 1 .0.
-// CHECK:STDOUT: (+) stack-push: 1 .0.
+// CHECK:STDOUT: (+) stack-push: pos-0 ValueExpressionAction `1` (full_trace.carbon:12)
+// CHECK:STDOUT: (+) stack-push: pos-0 ExpressionAction `1` (full_trace.carbon:12)
 // CHECK:STDOUT: --- step exp 1 .0. (full_trace.carbon:12) --->
-// CHECK:STDOUT: (-) stack-pop: 1 .0.
-// CHECK:STDOUT: (-) stack-pop: 1 .1. {{[[][[]}}1]]
+// CHECK:STDOUT: (-) stack-pop:  pos-0 ExpressionAction `1` (full_trace.carbon:12)
+// CHECK:STDOUT: (-) stack-pop:  pos-1 ValueExpressionAction `1` results: [`1`]  (full_trace.carbon:12)
 // CHECK:STDOUT: --- step exp (n + 1) .2. (full_trace.carbon:12) --->
-// CHECK:STDOUT: (-) stack-pop: (n + 1) .2. {{[[][[]}}0, 1]]
-// CHECK:STDOUT: (-) stack-pop: (n + 1) .1. {{[[][[]}}1]]
+// CHECK:STDOUT: (-) stack-pop:  pos-2 ExpressionAction `(n + 1)` results: [`0`, `1`]  (full_trace.carbon:12)
+// CHECK:STDOUT: (-) stack-pop:  pos-1 ValueExpressionAction `(n + 1)` results: [`1`]  (full_trace.carbon:12)
 // CHECK:STDOUT: --- step stmt return (n + 1); .1. (full_trace.carbon:12) --->
 // CHECK:STDOUT: +++ memory-write: #1 `1`
-// CHECK:STDOUT: (-) stack-pop: return (n + 1); .1. {{[[][[]}}1]]
-// CHECK:STDOUT: (-) stack-pop: {return (n + 1);} .1. {}
-// CHECK:STDOUT: (-) stack-pop: .0. {n: i32: 0}
-// CHECK:STDOUT: (+) stack-push: clean up.0. {n: i32: 0}
-// CHECK:STDOUT: (+) stack-push: clean up.0. {}
-// CHECK:STDOUT: (-) stack-pop: clean up.0. {}
-// CHECK:STDOUT: (-) stack-pop: clean up.0. {n: i32: 0}
+// CHECK:STDOUT: (-) stack-pop:  pos-1 StatementAction `return (n + 1);` results: [`1`]  (full_trace.carbon:12)
+// CHECK:STDOUT: (-) stack-pop:  pos-1 StatementAction `{return (n + 1);}` scope: [] (full_trace.carbon:13)
+// CHECK:STDOUT: (-) stack-pop:  pos-0 ScopeAction `` scope: [`n: i32`: `0`] (None)
+// CHECK:STDOUT: (+) stack-push: pos-0 CleanUpAction `clean up` scope: [`n: i32`: `0`] (stack cleanup:1)
```

Comment:
"\`clean up\`" seems redundant, now that the output already tells me it's a `CleanUpAction`.